sc/src/main/javascript/bundles/experiment-page/src/ExperimentPageRouter.js

```diff
@@ -1,29 +1,116 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import {BrowserRouter, Route} from 'react-router-dom'
-
-import ExperimentPageView from './ExperimentPageView'
-
-const ExperimentPageRouter = (props) =>
-  <BrowserRouter>
-    <div>
-      <Route path={`/`}
-             render={(componentProps) => (
-               <ExperimentPageView {...componentProps}
-                                   atlasUrl={props.atlasUrl}
-                                   resourcesUrl={props.resourcesUrl}
-                                   suggesterEndpoint={props.suggesterEndpoint}
-                                   experimentAccession={props.experimentAccession}
-                                   availableClusters={props.availableClusters} /> )} />
-    </div>
-  </BrowserRouter>
+import {BrowserRouter, Route, Redirect, Switch, NavLink, withRouter} from 'react-router-dom'
 
-ExperimentPageRouter.propTypes = {
+import URI from 'urijs'
+
+import TSnePlotViewRoute from './TSnePlotViewRoute'
+
+
+const RoutePropTypes = {
+  match: PropTypes.object.isRequired,
+  location: PropTypes.object.isRequired,
+  history: PropTypes.object.isRequired
+}
+
+const TabCommonPropTypes = {
   atlasUrl: PropTypes.string.isRequired,
-  resourcesUrl: PropTypes.string,
   experimentAccession: PropTypes.string.isRequired,
-  availableClusters: PropTypes.array.isRequired,
-  suggesterEndpoint: PropTypes.string
+  accessKey: PropTypes.string,
+  resourcesUrl: PropTypes.string
+}
+
+
+// What component each tab type should render, coupled to ExperimentController.java
+const tabTypeComponent = {
+  // 'multipart' : ``,
+  't-sne-plot' : TSnePlotViewRoute
+  // 'experiment-design' : `ExperimntDesign`,
+  // 'resources' : `Resources`,
+  // 'static-table' : StaticTable,
+  // 'qc-report' : QCReport
+}
+
+const TopRibbon = ({tabNames, routeProps}) =>
+  <ul className={`tabs`}>
+    {
+      tabNames.map((tabName) =>
+        <li title={tabName} key={tabName} className={`tabs-title`}>
+          <NavLink to={{pathname:`/${tabName}`, search: routeProps.location.search, hash: routeProps.location.hash}}
+                   activeStyle={{color: `#0a0a0a`, background: `#e6e6e6`}}>
+            {tabName}
+          </NavLink>
+        </li>
+    )}
+  </ul>
+
+TopRibbon.propTypes = {
+  tabNames: PropTypes.arrayOf(PropTypes.string).isRequired,
+  routeProps: PropTypes.shape(RoutePropTypes)
+}
+
+
+const TabContent = ({type, tabProps, commonProps, routeProps}) => {
+  // Pass in the search from location
+  const Tab = tabTypeComponent[type]
+
+  return (
+    Tab ? <Tab {...tabProps} {...commonProps} {...routeProps}/> : null
+  )
+}
+
+TabContent.propTypes = {
+  type: PropTypes.string.isRequired,
```

Comment:
What is type here?

Reply:
The type of the tab, it determines what the component that will be rendered in it. See https://github.com/gxa/atlas-experiment/blob/master/src/ExperimentContainer.js#L20 and `createPageSection` for a more complete example.